config-model/src/main/java/com/yahoo/searchdefinition/expressiontransforms/TensorFlowFeatureConverter.java

```diff
@@ -0,0 +1,114 @@
+package com.yahoo.searchdefinition.expressiontransforms;
+
+import com.yahoo.searchdefinition.RankProfile;
+import com.yahoo.searchlib.rankingexpression.evaluation.TensorValue;
+import com.yahoo.searchlib.rankingexpression.integration.tensorflow.ImportResult;
+import com.yahoo.searchlib.rankingexpression.integration.tensorflow.TensorFlowImporter;
+import com.yahoo.searchlib.rankingexpression.rule.Arguments;
+import com.yahoo.searchlib.rankingexpression.rule.CompositeNode;
+import com.yahoo.searchlib.rankingexpression.rule.ConstantNode;
+import com.yahoo.searchlib.rankingexpression.rule.ExpressionNode;
+import com.yahoo.searchlib.rankingexpression.rule.ReferenceNode;
+import com.yahoo.searchlib.rankingexpression.transform.ExpressionTransformer;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Replaces instances of the tensorflow(model-path, signature, output)
+ * pseudofeature with the native Vespa ranking expression implementing
+ * the same computation.
+ *
+ * @author bratseth
+ */
+public class TensorFlowFeatureConverter extends ExpressionTransformer {
+
+    private final TensorFlowImporter tensorFlowImporter = new TensorFlowImporter();
+    private final RankProfile profile;
+
+    public TensorFlowFeatureConverter(RankProfile profile) {
+        this.profile = profile;
+    }
+
+    @Override
+    public ExpressionNode transform(ExpressionNode node) {
+        if (node instanceof ReferenceNode)
+            return transformFeature((ReferenceNode) node);
+        else if (node instanceof CompositeNode)
+            return super.transformChildren((CompositeNode) node);
+        else
+            return node;
+    }
+
+    private ExpressionNode transformFeature(ReferenceNode feature) {
+        try {
+            if ( ! feature.getName().equals("tensorflow")) return feature;
+
+            if (feature.getArguments().isEmpty())
+                throw new IllegalArgumentException("A tensorflow node must take an argument pointing to " +
+                                                   "the tensorflow model directory under [application]/models");
+
+            // Find the specified expression
+            ImportResult result = tensorFlowImporter.importModel(asString(feature.getArguments().expressions().get(0)));
+            ImportResult.Signature signature = chooseOrDefault("signatures", result.signatures(),
+                                                               optionalArgument(1, feature.getArguments()));
+            String output = chooseOrDefault("outputs", signature.outputs(),
+                                            optionalArgument(2, feature.getArguments()));
+
+            // Add all constants
+            result.constants().forEach((k, v) -> profile.addConstantTensor(k, new TensorValue(v)));
+
+            return result.expressions().get(output).getRoot();
+        }
```

Comment:
We could consider validating that macros for the placeholders actually are present. Or issue a warning if not found?

Reply:
Yes, but I think we already do validate that during deployment.